pyth-rng/src/config.rs

```diff
@@ -47,31 +52,31 @@ pub enum Options {
     /// Generate a random number by running the entire protocol end-to-end
     Generate(GenerateOptions),
 
+    /// Get the status of a pending request for a random number.
     GetRequest(GetRequestOptions),
 }
 
 #[derive(Args, Clone, Debug)]
-#[command(next_help_heading = "Ethereum Options")]
-#[group(id = "Ethereum")]
-pub struct EthereumOptions {
-    /// A 20-byte (40 char) hex encoded Ethereum private key.
-    /// This key is required to submit transactions (such as registering with the contract).
-    #[arg(long = "private-key")]
-    #[arg(env = "PRIVATE_KEY")]
-    #[arg(default_value = None)]
-    pub private_key: Option<String>,
+#[command(next_help_heading = "Config Options")]
+#[group(id = "Config")]
+pub struct ConfigOptions {
+    /// A secret used for generating new hash chains. A 64-char hex string.
+    #[arg(long = "config")]
+    #[arg(env = "PYTH_CONFIG")]
+    #[arg(default_value = "config.yaml")]
+    pub config: String,
+}
 
-    /// URL of a Geth RPC endpoint to use for interacting with the blockchain.
-    #[arg(long = "geth-rpc-addr")]
-    #[arg(env = "GETH_RPC_ADDR")]
-    #[arg(default_value = "https://goerli.optimism.io")]
-    pub geth_rpc_addr: String,
+impl ConfigOptions {
+    pub fn load(&self) -> Result<Config, Box<dyn Error>> {
```

Comment:
It would be better to move this method to `Config`, as this is a constructor for `Config` in disguise:

```rust
impl Config {
    fn from_file(path: &PathBuf) -> anyhow::Result<Self> {
        let yaml_content = fs::read_to_string(&self.config)?;
        let config: Config = serde_yaml::from_str(&yaml_content)?;
        ...
    }
}
```

Which you can call later with `Config::from_file(opts.config)?;` where you currently do this.